### PR TITLE
use typeof as well as instanceof

### PR DESCRIPTION
**Pull request by samthor**
Use `typeof` in addition to `instanceof` to check underlying types, useful for proxying across different documents.

### CODE REVIEW

File: proxy.js [hunk 1]
```diff
@@ -24,12 +24,28 @@
   let lastRevokeFn = null;
 
   /**
+   * @param {*} o
+   * @return {boolean} whether this is probably an Object
+   */
+  function isObject(o) {
+    return o instanceof Object || (!!o && (typeof o == 'object' || typeof o == 'function'));
```
Comment (caridy) [inline proxy.js]: redundant, use:

```
return o && (typeof o == 'object' || typeof o == 'function');
```

File: proxy.js [hunk 2]
```diff
@@ -24,12 +24,28 @@
...
+  }
+
+  /**
+   * @param {*} o
+   * @return {boolean} whether this is probably a Function
+   */
+  function isFunction(o) {
+    return o instanceof Function || (!!o && typeof o == 'function');
```
Comment (caridy) [inline proxy.js]: redundant, use:

```
return typeof o === 'function';
```

File: proxy.js [hunk 3]
```diff
@@ -24,12 +24,28 @@
...
+   */
+  function isObject(o) {
+    return o instanceof Object || (!!o && (typeof o == 'object' || typeof o == 'function'));
+  }
+
+  /**
+   * @param {*} o
+   * @return {boolean} whether this is probably a Function
+   */
+  function isFunction(o) {
```
Comment (caridy) [inline proxy.js]: since this is used only once, I will recommend to remove this method, and just do a typeof inline

File: proxy.js [hunk 4]
```diff
@@ -24,12 +24,28 @@
...
+  }
+
+  /**
    * @constructor
    * @param {!Object} target
    * @param {{apply, construct, get, set}} handler
    */
   scope.Proxy = function(target, handler) {
-    if (!(target instanceof Object && handler instanceof Object)) {
+    if (!isObject(target) || !isObject(handler)) {
```
Comment (caridy) [inline proxy.js]: maybe just split this condition into two conditions, and just use the typeof check in each, not need to introduce `isObject()` abstraction.

File: proxy.js (hunk 1 above)
Comment (samthor) [inline proxy.js]: Done

File: proxy.js (hunk 3 above)
Comment (samthor) [inline proxy.js]: Done

File: proxy.js (hunk 2 above)
Comment (samthor) [inline proxy.js]: Done

File: proxy.js (hunk 4 above)
Comment (samthor) [inline proxy.js]: We need to check if this is an `object` or `function`, since both are valid, so I've left the method.

**Comment by samthor** [comment]
PTAL

**Comment by caridy** [comment]
+1

**Comment by caridy** [comment]
:beers:

**Pull request merged**